common/src/main/java/io/netty/util/concurrent/DefaultEventExecutorChooserFactory.java

```diff
@@ -32,6 +32,9 @@ private DefaultEventExecutorChooserFactory() { }
     @SuppressWarnings("unchecked")
     @Override
     public EventExecutorChooser newChooser(EventExecutor[] executors) {
+        if (executors.length == 1) {
```

Comment:
Yep, my suggestion for Netty 5 would be to just make it simpler using lambdas, given that the JVM trust captured final fields and this could make the code much simpler without perf losses